addons/crm/models/crm_lead.py

```diff
@@ -0,0 +1,1188 @@
+            'type': 'contact'
+        }
+        partner = self.env['res.partner'].create(vals)
+        return partner
+
+    def _convert_opportunity_data(self, customer, team_id=False):
+        self.ensure_one()
+        if customer:
+            customer.address_get()['contact']
+        if not team_id:
+            team_id = self.team_id.id
+        val = {
+            'planned_revenue': self.planned_revenue,
+            'probability': self.probability,
+            'name': self.name,
+            'partner_id': customer and customer.id or False,
+            'lead_type': 'opportunity',
+            'date_open': fields.Datetime.now(),
+            'email_from': customer and customer.email or self.email_from,
+            'phone': customer and customer.phone or self.phone,
+            'date_conversion': fields.Datetime.now(),
+        }
+        if not self.stage_id or self.stage_id.stage_type == 'lead':
+            val['stage_id'] = self.stage_find(team_id, domain=[('stage_type', 'in', ('opportunity', 'both'))])
+        return val
+
+    @api.model
+    def retrieve_sales_dashboard(self):
+        res = {
+            'meeting': {
+                'today': 0,
+                'next_7_days': 0,
+            },
+            'activity': {
+                'today': 0,
+                'overdue': 0,
+                'next_7_days': 0,
+            },
+            'closing': {
+                'today': 0,
+                'overdue': 0,
+                'next_7_days': 0,
+            },
+            'done': {
+                'this_month': 0,
+                'last_month': 0,
+            },
+            'won': {
+                'this_month': 0,
+                'last_month': 0,
+            },
+            'nb_opportunities': 0,
+        }
+
+        opportunities = self.search_read(
+            [('lead_type', '=', 'opportunity'), ('user_id', '=', self.env.uid)],
+            ['date_deadline', 'next_activity_id', 'date_action', 'date_closed', 'planned_revenue'])
+
+        for opp in opportunities:
+            # Expected closing
+            if opp['date_deadline']:
+                date_deadline = fields.Date.from_string(opp['date_deadline'])
+                if date_deadline == fields.Date.today():
+                    res['closing']['today'] += 1
+                if date_deadline >= fields.Date.today() and date_deadline <= fields.Date.today() + timedelta(days=7):
+                    res['closing']['next_7_days'] += 1
+                if date_deadline < fields.Date.today():
+                    res['closing']['overdue'] += 1
+
+            # Next activities
+            if opp['next_activity_id'] and opp['date_action']:
+                date_action = fields.Date.from_string(opp['date_action'])
+                if date_action == fields.Date.today():
+                    res['activity']['today'] += 1
+                if date_action >= fields.Date.today() and date_action <= fields.Date.today() + timedelta(days=7):
```

Comment:
is it working ? can't compare datetime.date to str. date_action is datetime date type and fields.Date.today() is string format  . you have to convert it in datetime date. then it's work.